tools/memap.py

```diff
@@ -340,7 +355,97 @@ def search_objects(self, path, toolchain):
                     else:
                         self.object_to_module.update({object_name:module_name})
 
-    export_formats = ["json", "csv-ci", "table"]
+    export_formats = ["json", "csv-ci", "table", "html"]
+
+    def memap_generate_html_module_report(self, module_data):
+        """! Function generates memory map usage HTML with bar chart and summary on the right
+        @param module_data Dict with module memory map usage statistics
+
+        module_data = {
+            "module": "Misc",
+            "size": {
+                ".data": 2228,
+                ".bss": 2152,
+                ".text": 57635
+            }
+        }
+        @return HTML formatted output with module's memory map stats
+        """
+
+        result = str()
+        module_size = module_data.get('size', module_data.get('summary', dict()))
+        table_str = """
+        <table cellspacing=5 cellpadding=0>
+            <tr>
+                <td>
+                    <table cellspacing=0 cellpadding=2>
+                      <tr>
+                        %s
+                      </tr>
+                    </table>
+                </td>
+                <td>
+                    <table cellspacing=2 cellpadding=0>
+                        %s
+                    </table>
+                </td>
+            </tr>
+        </table>
+        """
+
+        if module_size:
+            total_section_size = 0
+            for section in sorted(module_size):
+                # section: .data, .bss, .text
+                total_section_size += module_size[section]
+
+            td_str = str()
+            td_summary_str = str()
+            for section in sorted(module_size):
+                section_size = 100.0 * module_size[section] / total_section_size
+                td_size_px = 4 * int(section_size)
+                #print td_size_px, module_size[section], total_section_size
+                color = self.html_section_color_map.get(section, "Silver")
+                td_str += """
+                <td valign='top' bgcolor='%s' width='%dpx' height='55px' title='Section %s %d Bytes'>
+                </td>
+                """% (color,
+                    td_size_px,
+                    section,
+                    module_size[section])
+
+                td_summary_str += """
+                <tr>
+                    <td width=150px>Section <font color='%s'><b>%s</b></font>:</td>
```

Comment:
The `<font>` tag has been deprecated from the HTML specification for a long time: http://www.w3schools.com/tags/tag_font.asp

Can we switch to using CSS? The same thing is valid for the following HTML attributes: `width`, `height`, `align`, `bgcolor`